dataactcore/scripts/fixCountyData.py

```diff
@@ -0,0 +1,328 @@
+
+
+def update_fpds_ppop(sess):
+    logger.info("Starting FPDS PPOP derivations, starting FPDS PPOP 9-digit zips without dashes")
+
+    # FPDS PPOP 9-digit no dash
+    sess.execute(
+        """UPDATE detached_award_procurement AS dap
+            SET place_of_perform_county_co = zc.county_number,
+                place_of_perform_county_na = CASE WHEN dap.place_of_perform_county_na IS NOT NULL
+                                                  THEN dap.place_of_perform_county_na
+                                                  ELSE UPPER(zc.county_name) END
+            FROM zip_county AS zc
+            WHERE zc.combined_zip = dap.place_of_performance_zip4a
+                AND dap.place_of_perform_county_co IS NULL
+                AND UPPER(dap.place_of_perform_country_c) = 'USA'"""
+    )
+    sess.commit()
+
+    logger.info("Finished FPDS PPOP 9-digit zips without dashes, starting FPDS PPOP 9-digit zips with dashes")
+
+    # FPDS PPOP 9-digit dash
+    sess.execute(
+        """UPDATE detached_award_procurement AS dap
+            SET place_of_perform_county_co = zc.county_number,
+                place_of_perform_county_na = CASE WHEN dap.place_of_perform_county_na IS NOT NULL
+                                                  THEN dap.place_of_perform_county_na
+                                                  ELSE UPPER(zc.county_name) END
+            FROM zip_county AS zc
+            WHERE zc.dashed_zip = dap.place_of_performance_zip4a
+                AND dap.place_of_perform_county_co IS NULL
+                AND UPPER(dap.place_of_perform_country_c) = 'USA'"""
+    )
+    sess.commit()
+
+    logger.info("Finished FPDS PPOP 9-digit zips with dashes, starting FPDS PPOP 5-digit zips")
+
+    # FPDS PPOP 5-digit
+    sess.execute(
+        """UPDATE detached_award_procurement AS dap
+            SET place_of_perform_county_co = sc.county_number,
+                place_of_perform_county_na = CASE WHEN dap.place_of_perform_county_na IS NOT NULL
+                                                  THEN dap.place_of_perform_county_na
+                                                  ELSE UPPER(sc.county_name) END
+            FROM single_county AS sc
+            WHERE sc.zip5 = LEFT(dap.place_of_performance_zip4a, 5)
+                AND dap.place_of_perform_county_co IS NULL
+                AND dap.place_of_performance_zip4a ~ '^\d{5}(-?\d{4})?$'
+                AND UPPER(dap.place_of_perform_country_c) = 'USA'"""
+    )
+    sess.commit()
+    logger.info("Finished FPDS PPOP 5-digit zips, FPDS PPOP updates complete")
+
+
+def update_fabs_le(sess):
+    logger.info("Starting FABS legal entity derivations, starting FABS legal entity 9-digit zips")
+
+    # FABS LE 9-digit
+    sess.execute(
+        """UPDATE published_award_financial_assistance AS pafa
+            SET legal_entity_county_code = zc.county_number,
+                legal_entity_county_name = CASE WHEN pafa.legal_entity_county_name IS NOT NULL
```

Comment:
General Question: If we are only updating county names that are null, is there any checks if county name is incorrectly formatted?